app-web/src/utils/hooks.js

```diff
@@ -92,3 +111,55 @@ export const useImplicitAuth = () => {
   }, []);
   return user;
 };
+
+// returns if user is authenticated and the id token
+export const useAuthenticated = () => {
+  const [authenticated, setAuthenticated] = useState({ authenticated: false, idToken: null });
+  const { auth } = useContext(AuthContext);
+  useEffect(() => {
+    if (!isEmpty(auth)) {
+      const now = new Date();
+      const { exp } = auth.idToken.data;
+      // jwt times are in seconds, multiply by 1000 to convert into a date object
+      const expDate = new Date(exp * 1000);
+      // parse out auth.id_token and see if its still valid
+      if (moment(now).isBefore(moment(expDate))) {
+        setAuthenticated({ authenticated: true, token: auth.idToken.bearer });
+      }
+    }
+  }, [auth]);
+  return authenticated;
+};
+
+/**
+ * hook that performs a rocket chat search against the rocket gate apollo api
+ * @param {Boolean} authenticated
+ * @param {String} queryString the search string
+ * @param {Client} client the apollo client to query
+ * @returns {Object} {loading, results: <Array>}
+ */
+export const useRCSearch = (authenticated, queryString, client) => {
+  const { data, loading } = useQuery(ROCKET_GATE_QUERY, {
+    variables: {
+      queryString,
+    },
+    client,
+  });
+  const [results, setResults] = useState([]);
+  const [_loading, setLoading] = useState(true);
+  useEffect(() => {
+    setLoading(loading);
+    if (!authenticated || !queryString) {
+      setResults([]);
+    } else if (!_loading) {
+      setResults(data.search);
+    }
+
+    return () => {
+      setResults([]);
```

Comment:
so it doesn't matter is it's loaded, will it eventually return the setResults([]); ?